app/controllers/concerns/idv_step_concern.rb

```diff
@@ -7,4 +7,28 @@ module IdvStepConcern
     before_action :confirm_two_factor_authenticated
     before_action :confirm_idv_needed
   end
+
+  def confirm_user_has_completed_verify_step
+    return if verify_step_complete?
+
+    redirect_to idv_verify_info_url
+  end
+
+  def verify_step_complete?
+    idv_session.profile_confirmation && idv_session.resolution_successful
```

Comment:
If both of these need checking, can we collapse them into one property? Maybe that's for a separate PR after this one goes out.